mypy/plugin.py

```diff
@@ -379,3 +421,346 @@ def int_pow_callback(ctx: MethodContext) -> Type:
+                        converter = get_argument(rvalue, 'converter')
+                        convert = get_argument(rvalue, 'convert')
+                        if convert and converter:
+                            ctx.api.fail("Can't pass both `convert` and `converter`.", rvalue)
+                        elif convert:
+                            converter = convert
+                        if (converter
+                                and isinstance(converter, RefExpr)
+                                and converter.node
+                                and isinstance(converter.node, FuncBase)
+                                and converter.node.type
+                                and isinstance(converter.node.type, CallableType)
+                                and converter.node.type.arg_types):
+                            typ = converter.node.type.arg_types[0]
+
+                        # Does this even have to go in init.
+                        init = get_bool_argument(rvalue, 'init', True)
+
+                        # When attrs are defined twice in the same body we want to use
+                        # the 2nd definition in the 2nd location. So remove it from the
+                        # OrderedDict
+                        if name in own_attrs:
+                            del own_attrs[name]
+                        own_attrs[name] = Attribute(name, typ, attr_has_default, init, stmt)
+                    elif auto_attribs and typ and stmt.new_syntax and not is_class_var(lhs):
+                        # `x: int` (without equal sign) assigns rvalue to TempNode(AnyType())
+                        has_rhs = not isinstance(rvalue, TempNode)
+                        if name in own_attrs:
+                            del own_attrs[name]
+                        own_attrs[name] = Attribute(name, typ, has_rhs, True, stmt)
+
+        elif isinstance(stmt, Decorator):
+            # Look for attr specific decorators.  ('x.default' and 'x.validator')
+            remove_me = []
+            for func_decorator in stmt.decorators:
+                if (isinstance(func_decorator, MemberExpr)
+                        and isinstance(func_decorator.expr, NameExpr)
+                        and func_decorator.expr.name in own_attrs):
+
+                    if func_decorator.name == 'default':
+                        # This decorator lets you set a default after the fact.
+                        own_attrs[func_decorator.expr.name].has_default = True
+
+                    if func_decorator.name in ('default', 'validator'):
+                        # These are decorators on the attrib object that only exist during
+                        # class creation time.  In order to not trigger a type error later we
+                        # just remove them.  This might leave us with a Decorator with no
+                        # decorators (Emperor's new clothes?)
+                        # TODO: Any way to type-check these?
```

Comment:
If you want to type-check them, then you should not remove them at this stage., since type checking happens at a later stage.

Reply:
I think this may be actually a question of having good `attrs` stubs.

Reply:
The actual problem here is that the code looks like this:

```
@attr.s
class A:
   x: int = attr.ib()

   @x.default
   def foo(self):
      return ...
```

So as far as mypy is concerned that `x` is an `int`.  So I definitely have to remove the `x.default` decorator.  I imagine I could replace it with a different decorator but I'm not even sure what that would look like.  I'll update the comment so that it's not a question.